### PR TITLE
Revise

### PR DESCRIPTION
**Pull request by jkrvivian**

### CODE REVIEW

**Comment by jkrvivian** [comment]
@jserv , please specifically checked on keywords, Introduction (Contribution), and Related Work :smile:

File: paper.tex [hunk 1]
```diff
@@ -45,11 +45,11 @@
...
...d unsubscribe. However, unlike static data sets, the streaming data increases the importance and difficulties of dynamic data ownership and identity verification. Using certificate authorities (CAs) in such diverse nature for identity verification exposes vulnerabilities of central services, which eventually threaten the trust foundation of the entire system. Therefore, a trustless data trading infrastructure is required where the entities can trade, validate data ownership and data integrity without trusting any service or participants. Besides, consider handling the jobs and trading processes at the same time is too heavy for low-level IoT devices, an automated subscription procedures is also required. In this paper, we leverage usages of distributed ledger technologies (DLTs) to construct a decentralized and trusted data trading platform on top of the IoT brokered infrastructure. This approach can efficiently enhance the degree of transparency and scalability. The storage via an end-to-end encrypted message streams allows transmitting, accessing and validating data streams over distributed ledgers without authorities, and the trading process is automated through smart contracts.
```
Comment (howjmay) [inline paper.tex]: makes our lives simpler

Comment (howjmay) [inline paper.tex]: The real-time data exchange among devices are periodical

Comment (howjmay) [inline paper.tex]: increases the importance and difficulty

Comment (howjmay) [inline paper.tex]: which eventually threatens the trust foundation

**Review by howjmay — CHANGES_REQUESTED** [review]

File: paper.tex [hunk 2]
```diff
@@ -142,7 +146,7 @@ \subsection{Data subscription-based trading platform players}
...
+We adopt the brokered infrastructure for three reasons. Firstly, it inherits the benefits from the standard pub/sub model which is more efficient than the request/response model specifically in a large scale system. Furthermore, if either side is offline, the proceeding tasks have to stop and start over. With brokers, the unfinished work can be cached or accomplished. Second, a fully decentralized architecture has difficulties to brings data providers and consumers together, both of them need to reveal more sensitive information, such as IP address, in order to build the communication tunnel. The existence of brokers resolved the privacy issue, as brokers are the bridges that link data providers and consumers, participants can trade with minimum info like its identifier and public key. Third, as brokers running on machines in a stable network and electricity environment, participants on resource-constrained devices can offload the computation tasks to brokers that enables an proxy server implementation mentioned in Section~\ref{section:ta_endpoint}. With the end-to-end encryption trasmission protocol we used, brokers can perform verifiable checks without knowing the contents.
```
Comment (howjmay) [inline paper.tex]: with minimum information

Comment (howjmay) [inline paper.tex]: enables a proxy server implementation

File: paper.tex (hunk 1 above)
Comment (jkrvivian) [reply paper.tex]: I  think plural is ok.

File: paper.tex [hunk 3]
```diff
@@ -274,7 +283,7 @@ \subsection{Enable Automated Trading Process}
...
+The delegation should ensure the privacy of data providers, for instance, brokers are asked to attach messages to MAM without knowing the contents as well as record encryption keys to smart contracts. To record encryption key to smart contract, data provider compute the digest of sensitive contents with hash function, and send the digest as well as the signature to brokers. Broker can first verify the signature from data provider with the public key on DID document. If it's valid, then the broker uploads the key, otherwise the process is aborted. Note that the broker need to sign the messages he/she receives, which has been proven to be key uploaded by a broker recorded on the Product Contract. The key upload time is also restricted in order to avoid data provider uploading fake key frequently. If the data consumers find out the encryption key does not match the signature or can not decrypt data streams as expected, they can launch a refund to withdraw their funds. Fig.~\ref{fig:key_upload} demonstrate the key upload to Product Contract via a broker.
```
Comment (howjmay) [inline paper.tex]: data provider computes the digest of sensitive contents

Comment (howjmay) [reply paper.tex]: and sends the digest along with the signature to brokers

Comment (howjmay) [inline paper.tex]: If it's valid, then the broker uploads the key, otherwise the process will be aborted.

Comment (howjmay) [inline paper.tex]: Note that the broker needs to sign the messages

Comment (howjmay) [inline paper.tex]: The key uploading time is also restricted

Comment (howjmay) [inline paper.tex]: they can launch a poll for refund

Comment (howjmay) [inline paper.tex]: Fig.~\ref{fig:key_upload} demonstrates the key uploaded to Product Contract via a broker.

**Review by howjmay — CHANGES_REQUESTED** [review]

File: paper.tex (hunk 1 above)
Comment (howjmay) [reply paper.tex]: I think using singular would be more in consistent, but I am ok with both

File: paper.tex [hunk 4]
```diff
@@ -45,11 +45,11 @@
...
 \end{abstract}
 
 \begin{IEEEkeywords}
-Data Marketplace, decentralization, Distributed Ledger Technology
+subscription-based IoT data trading, decentralization, Distributed Ledger Technology
```
Comment (jserv) [inline paper.tex]: Minimize the terms for keywords. Keep "data trading."

Comment (jserv) [reply paper.tex]: Separate "subscription" and "data trading."

File: paper.tex (hunk 3 above)
Comment (jkrvivian) [reply paper.tex]: I'll use "vote"

File: paper.tex [hunk 5]
```diff
@@ -380,11 +390,11 @@ \subsubsection{End-to-End-Encryption}
...
+\subsubsection{Register with Tangle-accelerator}
+User with channel seed can acquire channel chain ownership. Seed is generated by unique hardware information on edge device, so it should be transmitted to tangle-accelerator which allows it to publish messages on MAM channel chain. Asymmetric encryption is utilizing during the process of exchanging seed. Digital signature should be taken at the same time to ensure data integrity. Once the seed has been successfully exchanged, edge device is successfully registered on the specific tangle-accelerator. Thus, seed will not be exposed during communication after an edge device is successfully registered.
 
 \subsubsection{Issues in End-to-End-Encryption}
-E2EE helps a lot for low-level edge device utilizing blockchain service. It dramatically reduces execution time encryption takes. However, we met a few challenges in the E2EE we proposed.
+E2EE helps a lot for low-level edge device utilizing blockchain service. It dramatically reduces execution time encryption takes. However, we met two challenges in the E2EE:
```
Comment (howjmay) [inline paper.tex]: we met two challenges in the E2EE process:

**Review by howjmay — CHANGES_REQUESTED** [review]

**Comment by jkrvivian** [comment]
@HowJMay , could you take a look at the remote PoW part in Section Evaluation - message publishment :smile:

**Pull request merged**